node/subsystem/src/messages.rs

```diff
@@ -102,6 +123,16 @@ pub enum CandidateValidationMessage {
 	),
 }
 
+impl CandidateValidationMessage {
+	/// If the current variant contains the relay parent hash, return it.
+	pub fn relay_parent(&self) -> Option<Hash> {
+		match self {
+			Self::ValidateFromChainState(CandidateDescriptor{ relay_parent, ..}, _, _) => Some(*relay_parent),
+			Self::ValidateFromExhaustive(_, _, CandidateDescriptor{ relay_parent, ..}, _, _) => Some(*relay_parent),
```

Comment:
This variant should be `None`, based on my intuition.

Reply:
We're going to want the `CandidateValidationSubsystem` to accept these `ValidateFromExhaustive` messages regardless of whether there is a job at that relay_parent. In fact, the expected use-case is that this will be called well after that point, because this is used in the dispute resolution process after-the-fact.

Reply:
IMO, at the message level, we should just return the relay parent if one exists; we don't know what future usage we'll come up with. If `CandidateValidationSubsystem` wants to ignore the relay parent for these messages, then the appropriate place will be in `ToJobTrait::relay_parent`, which will better be able to handle special-casing like that.

Reply:
Well, messages that are designed to be targeted to a specific relay-parent have one as a top-level member of the variant. The `CandidateValidationMessage` specifically does not. I am arguing that semantically, the relay-parent of the descriptor is not the same thing as the relay-parent of the message and in fact we do not expect it to be used that way.

Reply:
Ok, that makes sense to me. Implemented in 863d7334.